pkg/corerp/renderers/container/render.go

```diff
@@ -616,6 +633,58 @@ func (r Renderer) makeDeployment(
 	return outputResources, secretData, nil
 }
 
+// Function to convert from map[string]EnvironmentVariable to map[string]corev1.EnvVar
+func convertEnvVar(key string, env datamodel.EnvironmentVariable, options renderers.RenderOptions) (corev1.EnvVar, error) {
+	if env.Value != nil {
+		return corev1.EnvVar{Name: key, Value: *env.Value}, nil
+	} else {
+		// There are two cases to handle here:
+		// 1. The value comes from a kubernetes secret
+		// 2. The value comes from a radius resource id.
+
+		// If the value comes from a kubernetes secret, we'll reference it.
+		if strings.HasPrefix(env.ValueFrom.SecretRef.Source, "/") {
+			secretStore, ok := options.Dependencies[env.ValueFrom.SecretRef.Source].Resource.(*datamodel.SecretStore)
+			if !ok {
+				return corev1.EnvVar{}, fmt.Errorf("failed to find source in dependencies: %s", env.ValueFrom.SecretRef.Source)
+			}
+			// The format may be <namespace>/<name> or <name>
+			var name string
+			if strings.Contains(secretStore.Properties.Resource, "/") {
+				name = strings.Split(secretStore.Properties.Resource, "/")[1]
+			} else {
+				name = env.ValueFrom.SecretRef.Source
+			}
+
+			return corev1.EnvVar{
+				Name: key,
+				ValueFrom: &corev1.EnvVarSource{
+					SecretKeyRef: &corev1.SecretKeySelector{
+						LocalObjectReference: corev1.LocalObjectReference{
+							Name: name,
+						},
+						Key: env.ValueFrom.SecretRef.Key,
+					},
+				},
+			}, nil
+
+		} else {
+			return corev1.EnvVar{
+				Name: key,
+				ValueFrom: &corev1.EnvVarSource{
+					SecretKeyRef: &corev1.SecretKeySelector{
+						LocalObjectReference: corev1.LocalObjectReference{
+							Name: env.ValueFrom.SecretRef.Source,
```

Comment:
I think we can extract the name decision to a separate function. We can use it in other places too.